03-transactions.md

```diff
@@ -364,7 +366,10 @@ Each node offering a signature:
   - MUST round each output down to whole satoshis.
   - MUST subtract the fee given by `fee_satoshis` from the output to the funder.
   - MUST remove any output below its own `dust_limit_satoshis`.
-  - MAY eliminate its own output.
+  - if its own output is below the network's dust threshold (see the [dust limits section](#dust-limits)):
```

Comment:
I think it would be cleaner to also add a tlv to `closing_signed` to advertise "I removed my own output" to make this explicit instead of having to check the sig against two distinct txs. Thoughts?

Reply:
I think you may be misunderstanding my comment (it's probably misplaced, it's not related to this specific line but is more general).

As this paragraph previously mentioned, the sender of `closing_signed` can always chose to remove its own output from the tx it's signing (for whatever reason they want) which means the receiver must test the signature against potentially two versions of the closing tx.

I'm suggesting to add a tlv to make this explicit: when a node signs a `closing_tx`, if its own output has been removed from that tx, it says so in that tlv. This way the receiver doesn't need to potentially check two distinct txs against the signature, it knows what "version" of the `closing_tx` has been signed.

Here is a concrete example. Let's imagine that Alice and Bob both set `dust_limit` to 546 sats.
Alice's main output is 1 000 sats and Bob's main output is 10 000 sats.
She decides to be nice to miners and trims her output: the signature she sends in her `closing_signed` is for a tx with only Bob's output.

According to the spec, Bob should check the signature against two potential txs: one with Alice and Bob's outputs, and one with Bob's output only. I'm suggesting that Alice adds a tlv in her `closing_signed` message that says "I removed my own output, the tx I signed contains only yours". This way Bob doesn't need to check two txs, he knows exactly what tx the signature should be for.

Reply:
BOLT3

> MAY eliminate its own output.

Right, I forgot that point! Okay I agree with the proposal part to explicitly signal one's own output removal from `closing_tx` inclusion through a new tlv and that way having the receiver knowing exactly what tx the signature is for.

Reply:
Yeah seems we need a new TLV here as @t-bast mentions, as otherwise the node receiving the signature may just reject it as being invalid if it doesn't know to drop the output. One hacky way to handle this would be to have the node just try the diff combinations of the outputs being there (or not). IMO it's better to make it explicit though (along with a feature bit), as otherwise we'll see co-op close attempts break down in practice.